### PR TITLE
Feat/my account

### PR DESCRIPTION
**Pull request by DonnA98**
not a real PR, just the one before the freeze

### CODE REVIEW

**Comment by DonnA98** [comment]
> I appreciate the effort in the commit message. Unfortunately it does not serve its primary purpose: it does not explain the change, instead it lists the classes added, that can be relatively easy deduced from the diff itself.
> 
> The commit message should state the purpose of the commit, why you had to add all those classes and some xml files, what integral change of the application the commit embodies.
> 
> Could you, please, try to formulate a better commit message as a reply to this comment?

add activities so the buttons in myAccount can lead to something and add several test for myAccount

File: app/build.gradle [hunk 1]
```diff
@@ -35,8 +35,8 @@ android {
     }
     lintOptions {
         disable 'InvalidPackage' // Toothpick runtime includes references to javax.inject which have
-                                 // runtime retention even if they're not used since we use registries
-                                 // for injection instead of reflection.
+        // runtime retention even if they're not used since we use registries
```
Comment (roosemberth) [inline app/build.gradle]: Undo this change

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -4,24 +4,32 @@
...
+
+                <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
-        <activity android:name=".activities.RemoteServiceGreeterActivity"/>
-        <activity android:name=".activities.SelectRemoteServicesProviderActivity"/>
+        <activity android:name=".activities.RemoteServiceGreeterActivity" />
+        <activity android:name=".activities.SelectRemoteServicesProviderActivity" />
+        <activity android:name=".activities.MyAccountActivity" />
+        <activity android:name=".activities.PendingRequestActivity" />
```
Comment (roosemberth) [inline app/src/main/AndroidManifest.xml]: please sort

File: app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+//        past_games.setOnClickListener(view -> startActivity(new Intent(this, PastGamesActivity.class)));
+//        // listener to start hostedGamesActivity
+//        LinearLayout hosted_games = findViewById(R.id.launch_hosted_games_activity_layout);
+//        hosted_games.setOnClickListener(view -> startActivity(new Intent(this, HostedGamesActivity.class)));
+//        // listener to start pastHostedGamesActivity
+//        LinearLayout past_hosted_games = findViewById(R.id.launch_past_hosted_games_activity_layout);
+//        past_hosted_games.setOnClickListener(view -> startActivity(new Intent(this, PastHostedGamesActivity.class)));
+//        // listener to start profileActivity
+//        LinearLayout profile= findViewById(R.id.launch_profile_activity_layout);
+//        profile.setOnClickListener(view -> startActivity(new Intent(this, ProfileActivity.class)));
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java]: Please remove this unnecessary comments. If they're useful add a TODO or FIXME with an asignee

File: app/src/main/res/layout/activity_my_account.xml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        <!--android:src="@mipmap/ic_launcher"-->
+        <!--android:adjustViewBounds="true"-->
+        <!--android:scaleType="fitEnd"-->
+        <!--android:contentDescription="die"-->
+        <!--app:layout_constraintRight_toRightOf="@id/constraint_layout_HG"/>-->
+
+<!--</android.support.constraint.ConstraintLayout>-->
+
+<!--</LinearLayout>-->
+
```
Comment (roosemberth) [inline app/src/main/res/layout/activity_my_account.xml]: Please remove this useless comments. If they're usefull add a TODO/FIXME with an asignee and an explaination

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java [hunk 5]
```diff
@@ -38,9 +38,11 @@ public void onCreate() {
         super.onCreate();
 
         // Initialise Dependency Injection framework
-        Toothpick.setConfiguration(Configuration.forProduction().disableReflection().preventMultipleRootScopes());
+        Toothpick.setConfiguration(
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: Can you please undo this and change and fix your editor settings?

File: app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java [hunk 6]
```diff
@@ -38,9 +38,11 @@ public void onCreate() {
...
+                Configuration.forProduction().disableReflection().preventMultipleRootScopes());
         FactoryRegistryLocator.setRootRegistry(new ch.epfl.sweng.erpa.smoothie.FactoryRegistry());
-        MemberInjectorRegistryLocator.setRootRegistry(new ch.epfl.sweng.erpa.smoothie.MemberInjectorRegistry());
+        MemberInjectorRegistryLocator.setRootRegistry(
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/ErpaApplication.java]: Same here

File: app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name)
+        );
+    }
+
+    private List<Drawable> cutDownImagesList() {
+        ArrayList<Drawable> myFilterDrawableList = new ArrayList<>(myDrawablesList);
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java]: Can you please use something more functionnal?
- Make a stream out of account buttons (or text for that matter)
- Make a stream with all the images
- Use `com.annimon.stream.Stream.zip` to make a stream of pairs.
- Collect the resulting stream into a list of pairs
- Replace the image on the last element of the pair (since it corresponds to the profile)

File: app/src/main/java/ch/epfl/sweng/erpa/model/MyAccountButton.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
+package ch.epfl.sweng.erpa.model;
+
+import java.util.Objects;
+
+public class MyAccountButton {
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/model/MyAccountButton.java]: Please use lombok

File: app/src/main/java/ch/epfl/sweng/erpa/operations/UserProfileCoordinator.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import ch.epfl.sweng.erpa.model.UserProfile;
+
+import static ch.epfl.sweng.erpa.services.UserSignupService.createAccessToken;
+
+@Singleton
+public class UserProfileCoordinator implements InvocationHandler, DependencyCoordinator<UserProfile> {
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/operations/UserProfileCoordinator.java]: You can remove invocation handler, `DependencyCoordinator` extends `InvocationHandler`

File: app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+                        R.string.pastGamesText), PastGamesActivity.class, true, false),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.hostedGamesText), HostedGamesActivity.class, false, true),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.pastHostedGamesText), PastHostedGamesActivity.class, false, true),
+                new MyAccountButton(this.getApplicationContext().getResources().getString(
+                        R.string.profileText), ProfileActivity.class, true, true)
+        );
+        myDrawablesList = Arrays.asList(
+                this.getApplicationContext().getDrawable(R.drawable.ic_action_name),
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java]: Please make a variable and add it multiple times e.g. `v = this.getApplica....; IntStream.rangeClosed(1, 8).map(i -> v).collect...`

File: app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+    private List<Drawable> cutDownImagesList() {
+        ArrayList<Drawable> myFilterDrawableList = new ArrayList<>(myDrawablesList);
+        for (int i = myDrawablesList.size() - 2; i >= myAccountButtonList.size() - 1; --i) {
+            myFilterDrawableList.remove(i);
+        }
+        return myFilterDrawableList;
+    }
+
+    private ArrayList<MyAccountButton> filterButtonList(List<MyAccountButton> l,
```
Comment (roosemberth) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java]: Please mind my remark on line 88

File: app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+                new MyAccountButton(resources.getString(R.string.confirmedGamesText),
+                        ConfirmedGamesActivity.class, true, false),
+                new MyAccountButton(resources.getString(R.string.pastGamesText),
+                        PastGamesActivity.class, true, false),
+                new MyAccountButton(resources.getString(R.string.hostedGamesText),
+                        HostedGamesActivity.class, false, true),
+                new MyAccountButton(resources.getString(R.string.pastHostedGamesText),
+                        PastHostedGamesActivity.class, false, true)
+        );
+        myDrawablesList = Stream.rangeClosed(1, 8).map(i -> context.getDrawable(R.drawable.ic_action_name)).collect(
```
Comment (rykeuse) [inline app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java]: Why do you range from 1 to 8 and not 1 to 6 (because of the D6 images) or 1 to `myAccountButtonList.size()`?

Comment (rykeuse) [reply app/src/main/java/ch/epfl/sweng/erpa/activities/MyAccountActivity.java]: sure
